runtime/lua/vim/iter.lua

```diff
@@ -1052,6 +1120,23 @@ function M.map(f, src, ...)
   return Iter.new(src, ...):map(f):totable()
 end
 
+--- Flattens a |list-iterator|, un-nesting nested values
+---
+--- ```lua
+--- -- Equivalent to:
+--- vim.iter(src):flat():totable()
+--- ```
+---
+---@see |Iter:flat()|
+---
+---@param src table|function Table or iterator function to flatten
+---@param depth number|nil Depth to which the iterable should be flattened
+---                        (defaults to 1)
+---@return table
+function M.flat(src, depth, ...)
+  return Iter.new(src, ...):flat(depth):totable()
```

Comment:
```suggestion
function M.flatten(src, depth, ...)
  return Iter.new(src, ...):flatten(depth):totable()
```

Reply:
since the other members here are in the imperative voice, that probably makes more sense 👍